src/Altinn.App.Api/Extensions/AltinnAppBuilder.cs

```diff
@@ -0,0 +1,59 @@
+using System.Diagnostics.CodeAnalysis;
+using OpenTelemetry;
+using OpenTelemetry.Logs;
+
+namespace Altinn.App.Api.Extensions;
+
+/// <summary>
+/// Class for defining extensions to IWebHostBuilder for AltinnApps
+/// </summary>
+public sealed class AltinnAppBuilder : IDisposable
+{
+    private IOpenTelemetryBuilder? _openTelemetryBuilder;
+
+    /// <summary>
+    /// Logging configurator
+    /// </summary>
+    internal Action<OpenTelemetryLoggerOptions>? LoggingConfigurator { get; private set; }
+
+    /// <summary>
+    /// Builder
+    /// </summary>
+    public IOpenTelemetryBuilder OpenTelemetry =>
+        _openTelemetryBuilder
+        ?? throw new InvalidOperationException(
+            "OpenTelemetry is not enabled - set 'UseOpenTelemetry' to 'true' in appsettings to use OpenTelemetry"
+        );
+
+    /// <summary>
+    /// Experimental API for configuring OpenTelemetry logging.
+    /// In newer versions of OpenTelemetry, this API will be improved
+    /// </summary>
+    /// <param name="configure">configuration delegate</param>
+    /// <returns></returns>
+    [Experimental("ALTINN010")]
```

Comment:
We should consider an analyzer so that we can customize the diagnostic message here